### ISSUE TITLE
Search fails on columns in data set but not included in table

### ISSUE THREAD

**Issue by sandikbarr** [AllenFang/react-bootstrap-table#213]
The search function fails with error in v1.4.1/src/store/TableDataStore.js line 210, Cannot read property 'format' of undefined. I have a dataset where not all columns are included in the table. colInfos[key] from the row set is undefined, so it fails with error.

(Also a request: Since you are in very active development, it would be super helpful if you could distribute with source maps to help us help you.)

**Comment by AllenFang** [comment]
@sandikbarr, thanks your suggestions :) if interesting, you can join [gitter](https://gitter.im/AllenFang/react-bootstrap-table) :)

Anyway, about your problem, could please give me your examples code? because ```colInfos``` is created from [src/BootstrapTable.js](https://github.com/AllenFang/react-bootstrap-table/blob/master/src/BootstrapTable.js#L58) and it's according to your definition of  ```TableHeaderColumn```. So I think there's should not be any problem If not all columns are included in the table.

Thanks :+1:

**Comment by sandikbarr** [comment]
Hi. Here is a rough example showing the problem I am having: https://github.com/sandikbarr/show-me-what-you-got If you use the custom search control I added, it errors with "Cannot read property 'format' of undefined" (Sorry about all the linting errors, it's late here. Lmk if you have any trouble getting it running.)

**Comment by AllenFang** [comment]
Fixed it, check the ```v1.4.2```

**Comment by sandikbarr** [comment]
I updated my example at https://github.com/sandikbarr/show-me-what-you-got to v1.4.2. I'm seeing a lot of  "Uncaught TypeError: Cannot read property '__html' of undefined" errors when I use my custom filter. I also tried 1.4.3, and I saw the same errors when I used the clear button on my custom search. My custom filter is just calling the handleFilterData, and search is calling handleSearch. Let me know how I can help track this down. Thanks.
